snap/hooks/post-refresh

```diff
@@ -5,3 +5,19 @@ if [ ! -f "$SNAP_DATA/config/daemon.json" ]; then
     mkdir -p "$SNAP_DATA/config"
     cp "$SNAP/config/daemon.json" "$SNAP_DATA/config/daemon.json"
 fi
+
+# Ensure nvidia support setup correctly #
+if snapctl is-connected opengl ; then
+    if snapctl is-connected nvidia-core22:graphics-core22 ; then
+
+        # Generate the CDI config #
+        $SNAP/usr/bin/nvidia-ctk cdi generate --nvidia-ctk-path "$SNAP/usr/bin/nvidia-ctk" --output="$SNAP_DATA/etc/cdi/nvidia.yaml"
+
+        # Generate the dockerd runtime config #
+        $SNAP/usr/bin/nvidia-ctk runtime configure --runtime=docker --runtime-path "$SNAP/usr/bin/nvidia-ctk" --config "$SNAP_DATA/config/daemon.json"
+
+        # Restart dockerd to reflect any changes #
+        snap restart "$SNAP_NAME.dockerd"
```

Comment:
what are the conditions for regenerating this data, and what does it depend on? Should it be rebuild each time there is change to nvidia.ko kernel driver? userspace libraries? Each time dockerd starts?

On every boot, nvidia-assemble assembles a potentially new driver, and then modprobes a new config. Should we generate this upon each dockerd daemon start up? to ensure it is correct, and restarting dockerd gets us the right config?

I'm not at all familiar with this, but if it is quick and ephemeral, I'd rather do this in like runtime-dir and do it on every dockerd startup. If at all possible.